agent/src/util/config_action.rs

```diff
@@ -276,6 +289,28 @@ async fn handle_config_delete(
     Ok(())
 }
 
+/// Checks to see if the configuration needs to be recreated.
+/// At present, this just checks to see if the `.metadata.generation` has changed.
+/// The `.metadata.generation` value is incremented for all changes, except for changes to `.metadata` or `.status`.
+async fn should_recreate_config(
+    config: &Configuration,
+    config_map: ConfigMap,
+) -> Result<bool, anyhow::Error> {
+    let name = config.metadata.name.clone().unwrap();
+    let last_generation = config_map
+        .lock()
+        .await
+        .get(&name)
+        .ok_or_else(|| anyhow::anyhow!("Configuration {} not found in ConfigMap", name))?
+        .last_generation;
+
+    if config.metadata.generation == last_generation {
```

Comment:
is there a chance that we'd get updates out of order?  i'm wondering if we wouldn't want <= concept rather than ==?

Reply:
Good question! I'm not sure TBH. I can make that change though.

Reply:
FYI ... I've made this change so we'll only consider it modified if the generation is greater than the last generation we had seen.